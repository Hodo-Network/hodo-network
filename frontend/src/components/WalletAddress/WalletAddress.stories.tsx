import { Story, Meta } from "@storybook/react";

import { WalletAddress, WalletAddressProps } from "./WalletAddress";
import { ZERO_ADDRESS } from "../../constants";

export default {
  title: "Components/WalletAddress",
  component: WalletAddress,
} as Meta;

const Template: Story<WalletAddressProps> = (args) => (
  <WalletAddress {...args} />
);

export const Default = Template.bind({});
Default.args = {};

export const Address = Template.bind({});
Address.args = {
  address: ZERO_ADDRESS,
};
