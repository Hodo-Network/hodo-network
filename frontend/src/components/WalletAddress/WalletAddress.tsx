import React from "react";
import ReactLoading from "react-loading";
import { TEXT_ADDRESS } from "../../constants/text";

export interface WalletAddressProps {
  address: string | null | undefined;
  loading?: boolean;
}

export const WalletAddress: React.FC<WalletAddressProps> = ({
  address,
  loading,
}) => {
  return (
    <div>
      <h1 className='text-sm uppercase font-medium text-gray-500 dark:text-gray-400'>
        {TEXT_ADDRESS}
      </h1>
      {loading ? (
        <ReactLoading
          type='bubbles'
          color='currentColor'
          className='text-blue-600 dark:text-blue-500'
        />
      ) : address ? (
        <p className='mt-2 font-medium text-gray-900 dark:text-gray-200 overflow-hidden overflow-ellipsis'>
          {address}
        </p>
      ) : (
        <p className='mt-2 font-medium text-gray-900 dark:text-gray-200 overflow-hidden overflow-ellipsis'>
          Not connected
        </p>
      )}
    </div>
  );
};
