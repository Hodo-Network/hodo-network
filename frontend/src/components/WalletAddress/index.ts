import { WalletAddress } from './WalletAddress';
export { WalletAddress };
