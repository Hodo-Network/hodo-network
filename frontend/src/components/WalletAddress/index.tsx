import React from "react";
import { TEXT_ADDRESS } from "../../constants/text";

export default function WalletAddress({ address }: { address: string }) {
  return (
    <div>
      <h1 className='text-sm uppercase font-medium text-gray-500 dark:text-gray-400'>
        {TEXT_ADDRESS}
      </h1>
      <p className='mt-2 font-medium text-gray-900 dark:text-gray-200 overflow-hidden overflow-ellipsis'>
        {address}
      </p>
    </div>
  );
}
