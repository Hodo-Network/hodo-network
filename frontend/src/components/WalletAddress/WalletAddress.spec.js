import ReactDOM from "react-dom";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import WalletAddress from "./index";

afterEach(cleanup);

test("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<WalletAddress />, div);
});

test("matches snapshot", () => {
  const tree = renderer.create(<WalletAddress />).toJSON();
  expect(tree).toMatchSnapshot();
});
