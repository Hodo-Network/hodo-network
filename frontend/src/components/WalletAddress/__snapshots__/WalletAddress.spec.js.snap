// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <h1
    className="text-sm uppercase font-medium text-gray-500 dark:text-gray-400"
  >
    Address
  </h1>
  <p
    className="mt-2 font-medium text-gray-900 dark:text-gray-200 overflow-hidden overflow-ellipsis"
  />
</div>
`;
